packages/@lwc/engine-core/src/framework/__tests__/context.spec.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { describe, it, expect, vi, beforeAll, afterEach, afterAll } from 'vitest';
+import { setTrustedContextSet, setContextKeys } from '@lwc/shared';
+import { logWarnOnce } from '../../shared/logger';
+import { connectContext, disconnectContext } from '../modules/context';
+
+// Mock the logger to inspect console output during tests
+vi.mock('../../shared/logger', () => ({
+    logWarnOnce: vi.fn(),
+}));
```

Comment:
We have some [helper methods](https://github.com/salesforce/lwc/blob/16048aac65baae6acd66240b70c82dd29b468ef6/packages/%40lwc/integration-not-karma/helpers/console.js#L7) defined already for the loggers you could use.